cluster/addons/fluentd-gcp/fluentd-gcp-configmap.yaml

```diff
@@ -75,7 +75,19 @@ data:
     <match reform.**>
       @type record_reformer
       enable_ruby true
-      tag raw.kubernetes.${tag_suffix[4].split('-')[0..-2].join('-')}
+      <record>
+        # Extract local_resource_id from tag for 'k8s_container' monitored
+        # resource. The format is:
+        # 'k8s_container.<namespace_name>.<pod_name>.<container_name>'.
+        "logging.googleapis.com/local_resource_id" ${"k8s_container.#{tag_suffix[4].rpartition('.')[0].split('_')[1]}.#{tag_suffix[4].rpartition('.')[0].split('_')[0]}.#{tag_suffix[4].rpartition('.')[0].split('_')[2].rpartition('-')[0]}"}
+        # Rename the field 'log' to a more generic field 'message'. This way the
```

Comment:
For `gke_container` resource, we are sending `log` field as `text_payload` after `time`/`severity`/`stream` are extracted. For every other resources, we are sending `message` field as textPayload if it is the only field.

To assure the same behavior for `k8s_container`, we can either rename the `log` field to `message` in the configuration, or hardcode an exception in the fluent-plugin-google-cloud gem next to the `gke_container` resource.

I'm slightly leaning towards handling it via configuration instead of hardcoding it in the gem. The only issue is that when there are additional fields (other than `time`/`severity`/`stream`/`log`), the behavior might be different. For the `gke_container` resource, we seem to be dropping all other fields (That feel like a bug to me, isn't it?). For the `k8s_container` resource, we will keep the jsonPayload structure and retain the additional fields.

WDYT? @bmoyles0117 @igorpeshansky

Reply:
For `gke_container`, the implementation and the configuration on the Kubernetes side were carefully crafted to work together. This, as you observed, was very fragile.
The semantics you propose is simple and straightforward. We can document that the logging agent expects the text payload in a `message` field, and rename `log` to `message` in the configuration.

Reply:
+1 to what Igor said.